### ISSUE TITLE
BNS One MVP - Mike Cohen & Snazzy

### ISSUE THREAD

**Issue by Hero-Gamer** [stacksgov/decentralized-grants#34]

**Comment by Snazzy-888** [comment]
**1) Total grant amount:** $4,300

**2) Project Description:**
BNS One, the complete product suite for registering, renewing and managing your BNS assets (and more!).

The project is a dedicated BNS platform with requirements driven by the BNS community. Design and product development work is already well specified and thought through under the title BNS One. The priority, first phase or milestone, of the grant will be to implement the design, animating the user experience connecting the application to the back via existing and new clarity contracts calls. 

The focus will be to prioritise and deliver the high value features and fundamental BNS use cases. 

The second phase will finalise the front end and include secondary / nice to have features. A road map to continue the project would be drawn up in consultation with the community but could include, for example, decentralised governance mechanisms along the lines of stx.eco and the ecosystem dao project, bitcoin focused features such as purchasing BNS names directly with bitcoin.

**3) Impact:**
With over 310K names already registered on the BNS system there is an urgent need for tools which allow BNS holders to manage their names and portfolios.

We've seen the importance of naming systems to blockchain growth in the way ENS has become the de facto standard at the heart of the Ethereum ecosystem and clearly BNS is set to play the same role not only for Stackers but potentially for Bitcoiners as well - along with the market growth that this implies.

Decentralised identity is at the heart of the Stacks mission which started out, in the early days, entirely about decentralised names. Diagrams of Zooko's Triangle peppered the early messaging and the initial burnchain for the project was in fact Namecoin and not Bitcoin. BNS One is so central to the Stacks enterprise that the case for funding a dedicated portal is unquestionable.

**4) Project Timelines, Milestones, and Deliverables:**
The scope of this application is strictly limited to the delivery of the figma designs of BNS One. It will be delivered in the following milestones;

Kick-off payment: 0% - $0

Milestone 1: Dashboard - 30% - $1290 - August 18th 2024
Deliverable: 
- Landing Page & Dashboard: This will include delivery of the Dashboard screen and will include the initial infrastructure tasks like dns registry, hosting and building the common components such as page headers and footers. This includes possible UI/UX redesign/rework of screens in Figma.

Costs: 20 days

- Milestone 2: Registration - 30% - $1290 - September 1st 2024
Deliverable:
- Registration: Delivery of the Registration screens and function. Includes interactions with contracts to register, revoke and transfer names and dealing with the BNS migration strategy. This includes possible UI/UX redesign/rework of screens in Figma.

Costs: 20 days

- Milestone 3: Delivery of BNS One MVP - 40% - $1720 - September 15th 2024
Deliverable: 
Tried, tested and safe for community use.

Costs: 10 days

**5) Is / will this project be open-sourced?**
No

**6) ZeroAuthorityDAO profile link:**
https://zeroauthority.xyz/profiles/SPQNJ28CTBN4BX7SGZBKVAP8FRBXX759TEXQQCB2

**Comment by Snazzy-888** [comment]
Due to work commitments I have only recently completed the KYC process on the 14th August, so the project start date has been delayed slightly. 

The new milestone dates are: 

1. 22nd September
2. 6th October
3. 27th October

This has been discussed with Rocky (call on 25th August) and is feasible for him and his team.

**Comment by Snazzy-888** [comment]
Based on discussions with Rocky's team I've updated the milestones for **BNS One**:

**Updated Milestones** 

**Milestone 1: Design**

**Cost:** $1,540

Requirements:

- Redesign the UI/UX of the current BNS One app on Figma.
- Deliverables include the moodboard, branding guide, Figma styles, landing page, dashboard, registration page, and profile page.

**Milestone 2: Development**

Cost: $2,760 (includes $400 app & hosting costs, $500 toward clarity dev work).

Requirements:

- Development of the dashboard and registration screens with infrastructure tasks (DNS registry, hosting, headers/footers).
- Registration screens to handle interactions with contracts for registration, revocation, and transfer of names.
- Delivery of the tested MVP, ready for community use.

Total Cost for BNS One: $4,300

Please note: additional support may be required for smart contract clarity development. However, the completion of the second milestone is not contingent on this, as Rocky's development team may not equipped to handle it.

**Comment by Snazzy-888** [comment]
**Update:**

The BNS One MVP is on track, although there may need to be a slight adjustment to the 'Design Milestone' with the date moved until the end of the week, Sunday 13th for completion. If anything changes here I'll post another update.

The 'Development Milestone' however is on track to start from Weds 9th, even with the design delayed slightly.

**Comment by Snazzy-888** [comment]
**Update:**

The BNS One MVP - **Design Milestone** is now complete, please see the completed deliverable and release the grant funds at your earliest convenience, thank you!

[BNS One MVP - Figma Prototype](https://www.figma.com/proto/8HxNweV5tJXSvrMO4zR8ai/BNS-(Shared)?page-id=52%3A17&node-id=65-69383&node-type=frame&viewport=4196%2C9625%2C0.48&t=EjHuzJHgsAB46bpi-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=65%3A69383&show-proto-sidebar=1)

**Comment by Hero-Gamer** [comment]
Hi @Snazzy-888, 

For Milestone 1, I do not see these deliverable items in the Figma file
**moodboard, branding guide**
Please provide or comment

**Comment by Snazzy-888** [comment]
Hi @Hero-Gamer please see the following:
[Moodboard](https://www.figma.com/board/hFfvCdBkjDHplIz9MJibS6/BNS-Moodboard?node-id=0-1&t=lsfgEQfIVu153jmn-1)
[Branding Guide](https://www.figma.com/design/8HxNweV5tJXSvrMO4zR8ai/BNS-(Shared)?node-id=71-6867&node-type=canvas&t=ilDOI6PkxjNde2j0-0)

**Comment by Hero-Gamer** [comment]
Nice one @Snazzy-888 Thanks for providing all deliverables' links.

I can confirm Milestone 1 has been completed.
```
Milestone 1: Design - Weds 9th October

Cost: $1,540

Requirements:

    Redesign the UI/UX of the current BNS One app on Figma.
    Deliverables include the moodboard, branding guide, Figma styles, landing page, dashboard, registration page, and profile page.
```

Dear @shaktistacks can you help distribute the payment? 🙏 

@Snazzy-888 Are you still on track for Milestone 3?

**Comment by Snazzy-888** [comment]
> Based on discussions with Rocky's team I've updated the milestones for **BNS One**:
> 
> **Updated Milestones**
> 
> **Milestone 1: Design - Weds 9th October**
> 
> **Cost:** $1,540
> 
> Requirements:
> 
> * Redesign the UI/UX of the current BNS One app on Figma.
> * Deliverables include the moodboard, branding guide, Figma styles, landing page, dashboard, registration page, and profile page.
> 
> **Milestone 2: Development - Sun 3rd November**
> 
> **Cost:** $2,760 (includes $400 app & hosting costs, $500 toward clarity dev work).
> 
> Requirements:
> 
> * Development of the dashboard and registration screens with infrastructure tasks (DNS registry, hosting, headers/footers).
> * Registration screens to handle interactions with contracts for registration, revocation, and transfer of names.
> * Delivery of the tested MVP, ready for community use.
> 
> Total Cost for BNS One: $4,300
> 
> Please note: additional support may be required for smart contract clarity development. However, the completion of the second milestone is not contingent on this, as Rocky's development team may not equipped to handle it.

Apologies, Rocky's dev will require an extension for **Milestone 2: Development - for Sun 10th November.**

We may finish earlier than the 10th, but we'll need to see how we progress. The slight delay is due to finalising the backend requirements for the app and website.

**Comment by Snazzy-888** [comment]
Update:

The **BNS One MVP** - **Development Milestone** is now complete. This is the 'pre-production' app as Rocky's dev isn't able to complete the Clarity side of the development.

> "Please note: additional support may be required for smart contract clarity development. However, the completion of the second milestone is not contingent on this, as Rocky's development team may not equipped to handle it."

Please see the completed deliverable and release the grant funds at your earliest convenience, thank you!

[https://bns.one/](https://bns.one/)

**Comment by Hero-Gamer** [comment]
> Update:
> 
> The **BNS One MVP** - **Development Milestone** is now complete. This is the 'pre-production' app as Rocky's dev isn't able to complete the Clarity side of the development.
> 
> > "Please note: additional support may be required for smart contract clarity development. However, the completion of the second milestone is not contingent on this, as Rocky's development team may not equipped to handle it."
> 
> Please see the completed deliverable and release the grant funds at your earliest convenience, thank you!
> 
> https://bns.one/

Hi Snazzy, 

* Thanks for the update. Consider Milestone 2 approved. 
* Please continue to push for Clarity integration so mainnet it can be ready for community use as per generally intended.
* Please leverage other grantee's work like [PeaceLoveMusic](https://x.com/PeaceLoveMusicG)'s [DeOrganized](https://x.com/DeOrganizedBTC) media hub. His show now regularly pulls in 100, 200, 300+ people, he can help you get your words out. Would be nice to see you have an interview with him about BNS! Happy to help you connect

```
Milestone 2: Development - Sun 3rd November

Cost: $2,760 (includes $400 app & hosting costs, $500 toward clarity dev work).
```

Dear @shaktistacks, please kindly disburse final payment at your convenience. Tysm!
